etna/models/nn/deepstate/state_space_model.py

```diff
@@ -0,0 +1,445 @@
+        """
+        transition_coeff = torch.eye(self.latent_dim())
+        return transition_coeff.float()
+
+    def innovation_coeff(self, datetime_index: Tensor) -> Tensor:
+        """Innovation coefficient matrix.
+
+        Parameters
+        ----------
+        datetime_index:
+            Tensor with the index values.
+            Values should be from 0 to seasonal period.
+
+        Returns
+        -------
+        :
+            Innovation coefficient matrix.
+        """
+        return self.emission_coeff(datetime_index)
+
+    def generate_datetime_index(self, timestamps: np.ndarray) -> np.ndarray:
+        """Generate datetime index to use in the State Space Model.
+
+        Parameters
+        ----------
+        timestamps:
+            Array with timestamps.
+
+        Returns
+        -------
+        :
+            Datetime index for State Space Model.
+        """
+        seq_length = timestamps.shape[0]
+        return np.zeros(shape=(seq_length,))
+
+
+class LevelTrendSSM(LevelSSM):
+    """Class for Level-Trend State Space Model."""
+
+    def latent_dim(self) -> int:
+        """Dimension of the latent space.
+
+        Returns
+        -------
+        :
+            Dimension of the latent space.
+        """
+        return 2
+
+    def transition_coeff(self, datetime_index: Tensor) -> Tensor:
+        """Transition coefficient matrix.
+
+        Parameters
+        ----------
+        datetime_index:
+            Tensor with the index values.
+            Values should be from 0 to seasonal period.
+
+        Returns
+        -------
+        :
+            Transition coefficient matrix.
+        """
+        transition_coeff = torch.eye(self.latent_dim())
+        transition_coeff[0, 1] = 1
+        return transition_coeff.float()
+
+
+class SeasonalitySSM(LevelSSM):
+    """Class for Seasonality State Space Model."""
+
+    def __init__(self, num_seasons: int, timestamp_transform: Callable[[pd.Timestamp], int]):
+        """Create instance of SeasonalitySSM.
+
+        Parameters
+        ----------
+        num_seasons:
+            Number of seasons in the considered seasonality period.
+        """
+        self.num_seasons = num_seasons
+        self.timestamp_transform = timestamp_transform
+
+    def latent_dim(self) -> int:
+        """Dimension of the latent space.
+
+        Returns
+        -------
+        :
+            Dimension of the latent space.
+        """
+        return self.num_seasons
+
+    def emission_coeff(self, datetime_index: Tensor) -> Tensor:
+        """Emission coefficient matrix.
+
+        Parameters
+        ----------
+        datetime_index:
+            Tensor with the index values.
+            Values should be from 0 to seasonal period.
+
+        Returns
+        -------
+        :
+            Emission coefficient matrix.
+        """
+        emission_coeff = one_hot(datetime_index.squeeze(-1), num_classes=self.latent_dim())
```

Comment:
The same about device

Reply:
Device is defined by datetime_index here. I think, we don't have to change code.